### ISSUE TITLE
SetBenchmark Silent Default to Equity

### ISSUE THREAD

**Issue by jaredbroad** [QuantConnect/Lean#2153]
#### Expected Behavior
Ideally the benchmark is resolved as BTCUSD not Equity.
```
SetBenchmark("BTCUSD");
AddCrypto("BTCUSD", Resolution.Minute);
```

#### Actual Behavior
BTCUSD added as Equity; and the AddCrypto line ignored. It was quite confusing as in this case the price was always $0 as there is no BTCUSD Equity. 

#### Potential Solution
Show a message about the Equity assumption? Scan market hours db to get security type? 

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

**Comment by mchandschuh** [comment]
I think the major issue here is providing an overload that doesn't require a security type, or rather a full symbol object. If required a full symbol object I think the code above would throw because BTCUSD wasn't added yet, but if the order was reversed then it would succeed since it now exists in the `SymbolCache`.

As always, wary of potential heuristics to resolve things that are more easily resolved by the user specifying the correct data.

**Comment by jaredbroad** [comment]
The API will always require a level of design and heuristics to make it user-friendly. As the API designers, we walk that line of making it easy to use, flexible and requiring total specifications.

**Comment by jaredbroad** [comment]
There may be a pending issue here with the ordering of the added security and resolving benchmark as there is code in place to select the right ticker: 
`var security = Securities.FirstOrDefault(x => x.Key.Value == symbol).Value;`

**Comment by StefanoRaggi** [comment]
Issues with the call order of AddEquity and SetBenchmark should have been resolved in #2403
